AI/VCAI/VCAI.h

```diff
@@ -216,14 +214,14 @@ class DLL_EXPORT VCAI : public CAdventureAI
 	Goals::TSubgoal questToGoal(const QuestInfo & q);
 
 	void recruitHero(const CGTownInstance * t, bool throwing = false);
-	bool isGoodForVisit(const CGObjectInstance * obj, HeroPtr h, SectorMap & sm);
+	bool isGoodForVisit(const CGObjectInstance * obj, HeroPtr h, boost::optional<uint32_t> movementCostLimit = boost::none);
+	bool isGoodForVisit(const CGObjectInstance * obj, HeroPtr h, AIPath & path);

```

Comment:
mark this as const

Reply:
also `isGoodForVisit` need change `VCAI::isTileNotReserved `to be const, to make it you need f.e make there 2 addidionals getters so it will be look like that:
```
bool VCAI::isTileNotReserved(const CGHeroInstance * h, int3 t) const
{
...
		if(obj && vstd::contains(ai->getReservedObjs(), obj) && !vstd::contains(ai->getReservedHeroesMap().at(h), obj))
...
}
```